openfl/utilities/click_types.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Click types module."""
+
+import ipaddress
+import re
+
+
+import click
+
+
+class DomainParamType(click.ParamType):
+    """Domain Type for click arguments."""
+
+    name = 'DomainType'
+
+    @staticmethod
+    def is_fqdn(hostname: str) -> bool:
+        """https://en.m.wikipedia.org/wiki/Fully_qualified_domain_name."""
+        if not 1 < len(hostname) < 253:
+            return False
+
+        # Remove trailing dot
+        if hostname[-1] == '.':
+            hostname = hostname[:-1]
+
+        #  Split hostname into list of DNS labels
+        labels = hostname.split('.')
+
+        #  Define pattern of DNS label
+        #  Can begin and end with a number or letter only
+        #  Can contain hyphens, a-z, A-Z, 0-9
+        #  1 - 63 chars allowed
+        fqdn = re.compile(r'^[a-z0-9]([a-z-0-9-]{0,61}[a-z0-9])?$', re.IGNORECASE) # noqa 
+
+        # Check that all labels match that pattern.
+        return all(fqdn.match(label) for label in labels)
+
+    def convert(self, value, param, ctx):
+        """Validate value, if value is valid, return it."""
+        if not self.is_fqdn(value):
+            self.fail(f'{value} is not a valid domain name', param, ctx)
+        return value
+
+
+class IpAddressParamType(click.ParamType):
+    """IpAddress Type for click arguments."""
+
+    name = 'IpAddress type'
+
+    @staticmethod
+    def is_api_adress(address: str) -> bool:
+        """Validate ip-adress value."""
+        try:
+            _ = ipaddress.ip_address(address)
+            return True
+        except ValueError:
+            return False
+
+    def convert(self, value, param, ctx):
+        """Validate value, if value is valid, return it."""
+        if not self.is_api_adress(value):
+            self.fail(f'{value} is not a valid ip adress name', param, ctx)
+        return value
+
+
+Domain = DomainParamType()
+IpAdress = IpAddressParamType()
```

Comment:
Variables should be capitalized. Like it used in `click`
`click.STRING`
https://click.palletsprojects.com/en/8.0.x/api/#click.STRING

Reply:
@dmitryagapov , yes, but upper case uses only for standard python types
As I understood, for more specific types they use C++ google code style ([Pascal_case](https://en.wiktionary.org/wiki/Pascal_case) for types naming) https://google.github.io/styleguide/cppguide.html#Type_Names
For example: `FloatRange`, `click.DateTime` and `click.Path`
Actually it was upper case before 2 last commits - https://github.com/intel/openfl/pull/192/commits/23973e1961a9f5c549d99caa4da7dde932c195bd , 
but @aleksandr-mokrov didn't prefer upper case here =)

Reply:
There is no any contradictions. `FloatRange` and `IntRange` are classes that's why we are using PascalCase. `click.INT` and `click.FLOAT` are class instances that's why we are using capital letters. Here you create class instances so you should use capital letters. In case if you import `DomainParamType` and use it like `type=DomainParamType()` it will be also correct. But not mixing it.

Reply:
@aleksandr-mokrov , I agree with Dmitry, because it looks like global constant instances, can we use upper case?

Reply:
Yes, I agree with this approach too.